flips/20201125-bip-44-multi-account.md

```diff
@@ -0,0 +1,196 @@
+assigned during an on-chain transaction. Flow supports a many-to-many relationship
+between accounts and public keys; multiple public keys can be added to a single account, 
+and a single public key can be added to multiple accounts. 
+
+The Flow protocol maintains an index from `address => [publicKey]`, 
+but there is currently no on-chain reverse mapping from `publicKey => [address]`.
+
+Implementers of this standard will require a separate source of truth,
+the public key registry, to return the associated user address for a given public key. 
+This requirement is mentioned in the [related issues](#related-issues) section below.
+
+#### Account Key Index Alignment
+
+The Flow protocol assigns an index to each account key that is registered on chain.
+When discussing this proposal, Flow contributors considered a variation that would 
+require the path index to always align with the on-chain index for the corresponding
+account.
+
+For example, a user may create an account from a seed phrase and generate public keys
+at indices 0, 1 and 2. If added to the account in order, their on-chain indices would
+also be 0, 1, 2, resulting in a clear mapping from each on-chain key 
+to its corresponding derivation path.
+
+The authors of this proposal chose to omit this restriction after considering that,
+in many cases, a user may add additional public keys from other sources, thus breaking
+the perfect alignment between on-chain and key path indices. Even without outside keys,
+it is still possible for keys to be added out of order.
+
+However, wallet developers or users may still incorrectly assume that the path 
+index of a public key always matches its on-chain account key index.
+
+#### Potential Misuse of BIP 44
+
+One could argue that this is a misuse of the original BIP 44, BIP 43 and BIP 32
+standards based on the fact that Flow does not derive addresses directly 
+from cryptographic public keys.
+
+#### Overdependence on BIP 44
+
+This proposal may also promote an overdependence on the BIP 44 standard,
+especially for wallets that do not already implement it. 
+In some cases it may be better to use a new implementation or
+adaptation of BIP 32 that is better suited to the Flow account model.
+
+### Dependencies
+
+* Dependent projects: Flow Ledger application, Flow Port
+
+### Tutorials and Examples
+
+TODO: add examples 
+
+### Compatibility
+
+This proposal does not change any existing APIs. 
+The chosen path structure is meant to maximize forward compatibility 
+in a variety of use cases.
+
+The convention proposed in this document is intended for use by any wallet
+that supports the BIP 44 standard.
+
+In particular, this proposal impacts wallet software that is compatible
+with the Flow Ledger application, such as Flow Port.
+
+## Related Issues
+
+### Mapping Public Keys to Addresses
+
+As mentioned in the [drawbacks](#drawbacks) section, this proposal necessitates 
+a public key registry that maps public keys to account addresses.
```

Comment:
```suggestion
a public key registry that maps public keys to account addresses. The information required
to associate Flow addresses with their keys is broadcast when keys are added to Flow
accounts both during and after account creation. Making this information searchable
requires indexing this information rather than requiring any additional disclosure on the
part of Flow account owners.
```